source/simulator/helper_functions.cc

```diff
@@ -1045,9 +1046,12 @@ namespace aspect
                                                                 introspection.component_indices.compositional_fields[advection_field.compositional_variable]
                                                                ),
                                                                /*dof index within component=*/ j);
-                    const double solution_value = distributed_solution(local_dof_indices[support_point_index]);
-                    const double limited_solution_value = theta * (solution_value-local_solution_average) + local_solution_average;
-                    distributed_solution(local_dof_indices[support_point_index]) = limited_solution_value;
+                    if (system_index_set.is_element(support_point_index))
```

Comment:
You are entirely correct -- for DG elements, every DoF on a locally owned cell is locally owned.

The if-statement you have here is wrong, and this is probably why you get different results: it would need to be
```
  if (system_index_set.is_element(local_dof_indices[support_point_index])
```
But, given your insight, there is no need for this any more. Can you just remove these changes again (e.g., by just dropping the commit, or by manually undoing it) and adding a comment such as "note that we are using DG elements, so every DoF on a locally owned cell is locally owned; this means that we do not need to check whether the 'distributed_solution' vector actually stores the element we read from/write to here"?